File: C/bfs.c
#include <stdio.h>
void BFS (int a[10][10],int n,int src,int s[10])
{
    int f=0,r=-1,q[20],i,v;
    printf ("Source Node: %d\n",src);
    s[src] = 1;
    q[++r] = src;
    printf("Visited Nodes using BFS:\n");
    while (f <= r)
    {
        v = q[f++];
        for (i=1;i<=n;i++)
        if (s[i]==0 && a[v][i])
        {
            q[++r] = i;
            printf ("%d\t",i);
            s[i] = 1;
        }
    }
}
int main ()
{
    int a[10][10],n,i,j,s[20],src;
    printf ("Enter no. of Nodes in Graph\n");
    scanf ("%d",&n);
    printf ("Enter the connections of Graph in Adjacency Matrix\n");
    for (i=1;i<=n;i++)
    for (j=1;j<=n;j++)
    scanf ("%d",&a[i][j]);
    printf ("Enter the Source Node\n");
    scanf ("%d",&src);
    for(i=1;i<=n;i++)
    s[i]=0;
    BFS (a,n,src,s);
    printf ("\n");
    return 0;
}

File: Python/BFS.py
import queue

# Syntax pada python untuk pembuatan graf menggunakan adjacency list
# graph = {
#     'A' : ['B','C'],
#     'B' : ['D','E'],
#     'C' : ['F'],
#     'D' : [],
#     'E' : ['F'],
#     'F' : ['H'],
#     'G' : ['H'],
#     'H' : ['G']
# }
# graph = {
#     'A' : ['B','F','E'],
#     'B' : ['G'],
#     'C' : ['D'],
#     'D' : ['H'],
#     'E' : ['F'],
#     'F' : ['B'],
#     'G' : ['C','H'],
#     'H' : ['D']
# }
graph = {
    'A' : ['B'],
    'B' : ['F','G'],
    'C' : ['D'],
    'D' : ['H'],
    'E' : ['F'],
    'F' : ['E'],
    'G' : ['C'],
    'H' : ['D']
}

# Buat Array untuk menyimpan node yang dikunjungi dan menyimpan antrian.
visited = [] # List to keep track of visited nodes.
queue = [] #Initialize a queue
# Buat fungsi BFS dengan menggunakan queue.
def bfs(visited, graph, node, goal):
    visited.append(node)
    queue.append(node)

    while queue:
        s = queue.pop(0)
        print(s, end = " ")
        if(s != goal):
            for neighbour in graph[s]:
                if neighbour not in visited:
                    visited.append(neighbour)
                    queue.append(neighbour)
        else:
            print("Selesai")
            break
# Tambahkan kode pemanggilan fugsi BFS.
print("Hasil penulusuran graf menggunakan BFS:")
bfs(visited, graph,'A','D')